tests/common/libtest.sh

```diff
@@ -189,6 +189,19 @@ setup_test_repository () {
     cd $oldpwd
 }
 
+run_temp_webserver() {
+    env PYTHONUNBUFFERED=1 setsid python -m SimpleHTTPServer 0 >${test_tmpdir}/httpd-output &
+    for x in $(seq 50); do
+        sed -e 's,Serving HTTP on 0.0.0.0 port \([0-9]*\) \.\.\.,\1,' < ${test_tmpdir}/httpd-output > ${test_tmpdir}/httpd-port
```

Comment:
Hmm, that's kinda icky, but yeah we don't want to hardcode a port either just because of that.